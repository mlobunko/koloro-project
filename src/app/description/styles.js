import styled from "styled-components";

export const StyledBoldParagraph = styled.div`
  width: 100%;
  text-align: center;
  font-weight: 700;
  max-width: 45rem;
  margin: 0 auto;
  padding: 2rem 1rem 0 1rem;

  @media (min-width: 1024px) {
    margin: ${props => (props.landscape ? "0" : "0 auto")};
    text-align: ${props => (props.landscape ? "left" : "center")};
    padding: ${props =>
      props.landscape ? "2rem 4rem 0 4rem" : "2rem 1rem 0 1rem"};
  }
`;
StyledBoldParagraph.displayName = "StyledBoldParagraph";

export const StyledH1 = styled.h1`
  width: 100%;
  text-align: center;
  margin: 0 auto;
  padding: 10rem 1rem 0 1rem;
  max-width: 53rem;

  @media (min-width: 1024px) {
    margin: ${props => (props.landscape ? "0" : "0 auto")};
    width: ${props => (props.landscape ? "70%" : "100%")};
    text-align: ${props => (props.landscape ? "left" : "center")};
    padding: ${props =>
      props.landscape ? "10rem 4rem 0 4rem" : "10rem 1rem 0 1rem"};
  }
`;
StyledH1.displayName = "StyledH1";

export const StyledItalicText = styled.span`
  font-style: italic;
`;
StyledItalicText.displayName = "StyledItalicText";

export const StyledStandardParagraph = styled.div`
  width: 100%;
  max-width: 45rem;
  text-align: center;
  font-style: normal;
  margin: 0 auto;
  padding: 2rem 1rem 0 1rem;

  @media (min-width: 1024px) {
    margin: ${props => (props.landscape ? "0" : "0 auto")};
    text-align: ${props => (props.landscape ? "left" : "center")};
    padding: ${props =>
      props.landscape ? "2rem 4rem 0 4rem" : "2rem 1rem 0 1rem"};
  }
`;
StyledStandardParagraph.displayName = "StyledStandardParagraph";
