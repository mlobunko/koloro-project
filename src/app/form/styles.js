import styled from "styled-components";

export const StyledButton = styled.button.attrs({
  type: props => props.type,
  "aria-label": props => props.ariaLabel
})`
  margin: 2rem 0;
  width: 100%;
  background-color: white;
  color: black;
  font-weight: 700;
  font-size: 0.8rem;
  padding: 0.8rem 4rem 0.8rem 4rem;
  border: none;
  border-radius: 2rem;

  &:hover {
    background-color: gold;
  }
  &:active {
    background-color: gold;
  }
`;
StyledButton.displayName = "StyledButton";

export const StyledForm = styled.form`
  width: 100%;
  max-width: 26.5rem;
  padding: 2rem 1rem 0 1rem;
  margin: 0 auto;
  text-align: center;

  @media (min-width: 1024px) {
    margin: ${props => (props.landscape ? "0" : "0 auto")};
    padding: ${props =>
      props.landscape ? "2rem 4rem 0 4rem" : "2rem 1rem 0 1rem"};
  }
`;
StyledForm.displayName = "StyledForm";

export const StyledInput = styled.input.attrs({
  type: props => props.type,
  placeholder: props => props.placeHolder
})`
  background: rgba(0, 0, 0, 0);
  width: 100%;
  border: none;
  border-bottom: 1px solid white;
  padding: 1rem 1rem 0.5rem 0;
  color: white;
  font-style: normal;
  font-weight: 400;

  &::placeholder {
    color: white;
    font-style: normal;
  }
`;
StyledInput.displayName = "StyledInput";
