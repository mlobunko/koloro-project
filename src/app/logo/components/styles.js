import styled from "styled-components";
import ReactTilt from "react-universal-tilt";

import koloroLogoO from "../images/koloroLogoO.png";
import koloroLogoWords from "../images/koloroLogoWords.png";

export const StyledLogoO = styled.div`
  position: absolute;
  height: 4rem;
  width: 11rem;
  top: 2rem;
  left: 0;
  right: 0;
  margin: auto;
  z-index: 2;
  background: url(${koloroLogoO}) no-repeat center center / contain;

  @media (min-width: 1024px) {
    margin-left: ${props => (props.landscape ? "4rem" : "auto")};
  }
`;
StyledLogoO.displayName = "StyledLogoO";

export const StyledLogoWords = styled.div`
  position: absolute;
  height: 4rem;
  width: 11rem;
  top: 2rem;
  left: 0;
  right: 0;
  margin: auto;
  z-index: 1;
  background: url(${koloroLogoWords}) no-repeat center center / contain;

  @media (min-width: 1024px) {
    margin-left: ${props => (props.landscape ? "4rem" : "auto")};
  }
`;

export const StyledReactTilt = styled(ReactTilt)`
  position: absolute;
  height: 4rem;
  width: 11rem;
  left: 0;
  right: 0;
  margin: auto;

  @media (min-width: 1024px) {
    margin-left: ${props => (props.landscape ? "0" : "auto")};
  }
`;

StyledLogoWords.displayName = "StyledLogoWords";
