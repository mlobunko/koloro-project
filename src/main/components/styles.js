import styled, { injectGlobal, keyframes } from "styled-components";
import PropTypes from "prop-types";

import imageBgRadiantLeft from "../images/imageBgRadiantLeft.png";
import imageBgRadiantRight from "../images/imageBgRadiantRight.png";
import imageGameOfThrones from "../images/imageGameOfThrones.png";
import imageGameOfThronesMini from "../images/imageGameOfThronesMini.png";
import fontGothamProLight from "../fonts/GothamPro-Light.ttf";
import fontGothamProBold from "../fonts/GothamPro-Bold.ttf";
import fontGothamProItalic from "../fonts/GothamPro-Italic.ttf";

injectGlobal`
  *, *:before, *:after {
    box-sizing: border-box;
  }
  html {
    font-family: Helvetica, Arial, sans-serif;
    font-weight: 400;
    
    @media (min-width: 2000px) {
      font-size: 150%;
    }
  }
`;

const animGradient = keyframes`
    0% {
        background-position: 0% 50%
    }
    50% {
        background-position: 100% 50%
    }
    100% {
        background-position: 0% 50%
    }
`;

export const StyledGradient = styled.div`
  background-color: #efcb19;
  background: linear-gradient(to right, #8141a9 0%, #d4436c 50%, #efcb19 100%);
  background-size: 500% 500%;
  width: 100%;
  min-height: 100vh;
  animation: ${animGradient} 30s linear infinite;
`;
StyledGradient.displayName = "StyledGradient";

export const StyledImageGameOfThrones = styled.div`
  width: 100%;
  height: 80vh;
  background: url(${imageGameOfThronesMini}) no-repeat bottom center/contain;

  @media (min-width: 1024px) {
    display: ${props => (props.landscape ? "none" : "block")};
  }
`;
StyledImageGameOfThrones.displayName = "StyledImageGameOfThrones";

StyledImageGameOfThrones.propTypes = {
  landscape: PropTypes.bool.isRequired
};

export const StyledMain = styled.div`
  display: flex;
  flex-direction: column;
  background: url(${imageBgRadiantLeft}) no-repeat center center / cover,
    url(${imageBgRadiantRight}) no-repeat center center / cover;
  width: 100%;
  color: white;
  font-family: GothamPro;
  font-weight: 400;
  min-height: 100vh;

  @media (min-width: 1024px) {
    display: block;
    background: ${props =>
        props.landscape
          ? `url(${imageGameOfThrones}) no-repeat center center/cover`
          : ``},
      url(${imageBgRadiantLeft}) no-repeat center center / cover,
      url(${imageBgRadiantRight}) no-repeat center center / cover;
  }

  @font-face {
    font-family: GothamPro;
    src: url(${fontGothamProLight});
    font-weight: 400;
    font-style: normal;
  }

  @font-face {
    font-family: GothamPro;
    src: url(${fontGothamProBold});
    font-weight: 700;
    font-style: normal;
  }

  @font-face {
    font-family: GothamPro;
    src: url(${fontGothamProItalic});
    font-weight: 400;
    font-style: italic;
  }
`;
StyledMain.displayName = "StyledMain";

StyledMain.propTypes = {
  landscape: PropTypes.bool.isRequired
};

export const StyledTextBlock = styled.div`
  width: 100%;
`;
StyledTextBlock.displayName = "StyledTextBlock";
