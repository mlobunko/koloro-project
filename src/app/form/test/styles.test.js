import React from "react";
import { shallow } from "enzyme";
import "enzyme-to-json";

import { StyledButton, StyledForm, StyledInput } from "../styles";

describe("StyledButton", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<StyledButton />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("StyledForm", () => {
  it("renders correctly, landscape true", () => {
    const landscape = true;
    const wrapper = shallow(<StyledForm landscape={landscape} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("renders correctly, landscape false", () => {
    const landscape = false;
    const wrapper = shallow(<StyledForm landscape={landscape} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("StyledInput", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<StyledInput />);
    expect(wrapper).toMatchSnapshot();
  });
});
