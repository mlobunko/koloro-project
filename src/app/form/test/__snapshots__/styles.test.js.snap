// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledButton renders correctly 1`] = `
<button
  className="sc-bdVaJa jlfEA"
/>
`;

exports[`StyledForm renders correctly, landscape false 1`] = `
<form
  className="sc-bwzfXH cdQIts"
/>
`;

exports[`StyledForm renders correctly, landscape true 1`] = `
<form
  className="sc-bwzfXH hdvaBa"
/>
`;

exports[`StyledInput renders correctly 1`] = `
<input
  className="sc-htpNat krZMaS"
/>
`;
